packages/node/src/serverless-functions/serverless-handler.mts

```diff
@@ -103,43 +86,40 @@ export async function createServerlessEventHandler(
 
   return async function (request: IncomingMessage) {
     const url = new URL(request.url ?? '/', server.url);
-    const response = await fetch(url, {
+    const response = await undiciRequest(url, {
       body: await serializeBody(request),
-      compress: !isStreaming,
-      // @ts-expect-error
       headers: {
         ...request.headers,
         host: request.headers['x-forwarded-host'],
       },
-      method: request.method,
-      redirect: 'manual',
+      method: (request.method || 'GET') as Dispatcher.HttpMethod,
     });
 
-    let body: Buffer | null = null;
-    let headers: Headers = response.headers;
+    let body: Readable | Buffer | null = null;
+    let headers = toHeaders(response.headers) as Headers;
 
     if (isStreaming) {
       body = response.body;
     } else {
-      body = Buffer.from(await response.arrayBuffer());
-
-      const contentEncoding = response.headers.get('content-encoding');
-      if (contentEncoding) {
-        body = compress(body, contentEncoding);
-        const clonedHeaders = [];
-        for (const [key, value] of response.headers.entries()) {
-          if (key !== 'transfer-encoding') {
-            // transfer-encoding is only for streaming response
-            clonedHeaders.push([key, value]);
-          }
-        }
-        clonedHeaders.push(['content-length', String(Buffer.byteLength(body))]);
-        headers = new Headers(clonedHeaders);
-      }
+      body = Buffer.from(await response.body.arrayBuffer());
+      // console.log('body', body.toString('utf-8'));
```

Comment:
<!-- Questions comments seek clarification. -->
**question (non-blocking):** ‏Is there any reason to keep this commented out block?

Reply:
just removed :)